### PR TITLE
chore: upgrade packages

### PR DESCRIPTION
**Pull request by tarashaliuk**
Updated dependencies:
- @storybook/jest from 0.0.10 to 0.2.3 (done)
- @nrwl/storybook from 14.5.8 to 16.9.1 (done)
- stylelint from 14.16.1 to 15.10.3 (done)
- word-wrap from 1.2.3 to 1.2.4 (not exists)
- eslint-plugin-prettier from 3.4.1 to 5.0.0 (done)
- glob-parent  from 3.1.0 to 5.1.2 (not exists)
- webpack from 4.46.0 to 5.77.0 (done)
- @storybook/web-components 6.5.16 to 7.0.2 (done)

partially closes: https://spryker.atlassian.net/browse/HRZ-89933

### CODE REVIEW

File: libs/base/ui/overlays/modal/src/stories/interactive/close-modal.stories.ts [hunk 1]
```diff
@@ -31,5 +31,5 @@ CloseModal.play = async (obj: {
   userEvent.click(closeBtn);
   await new Promise((r) => setTimeout(r, 1000));
 
-  expect(modal.hasAttribute('open')).toBeFalsy();
+  await expect(modal.hasAttribute('open')).toBeFalsy();
```
Comment (supproduction) [inline libs/base/ui/overlays/modal/src/stories/interactive/close-modal.stories.ts:31]: Can we use waitFor here?

File: libs/base/ui/overlays/modal/src/stories/interactive/close-nested-modal.stories.ts [hunk 2]
```diff
@@ -33,5 +33,5 @@ CloseNestedModal.play = async (obj: {
   userEvent.click(closeNestedModalBtn);
   await new Promise((r) => setTimeout(r, 1000));
 
-  expect(nestedModal.hasAttribute('open')).toBeFalsy();
+  await expect(nestedModal.hasAttribute('open')).toBeFalsy();
```
Comment (supproduction) [inline libs/base/ui/overlays/modal/src/stories/interactive/close-nested-modal.stories.ts:34]: Can we use waitFor here?

Comment (supproduction) [reply libs/base/ui/overlays/modal/src/stories/interactive/close-nested-modal.stories.ts:34]: Or can we omit it from other tests and use just await?

File: migrations.json
Comment (supproduction) [inline migrations.json:1]: Do we need it?

File: nx.json [hunk 3]
```diff
@@ -23,17 +15,18 @@
           "stylelint",
           "paths"
         ],
-        "parallel": 1
+        "parallel": 1,
+        "accessToken": "OGI1YjE1ZTMtMDJhMi00MDVlLTkwNWItMWViY2ZjNmNmZTFjfHJlYWQtd3JpdGU="
```
Comment (supproduction) [inline nx.json]: Do we need this token? Where ti comes from?

File: package.json [hunk 4]
```diff
@@ -66,73 +66,80 @@
...
-    "@nrwl/workspace": "15.7.2",
+    "@nrwl/js": "^16.0.0",
+    "@nx/cypress": "^16.0.0",
+    "@nx/devkit": "^16.0.0",
+    "@nx/eslint-plugin": "^16.0.0",
+    "@nx/js": "^16.0.0",
+    "@nx/linter": "^16.0.0",
+    "@nx/storybook": "^16.0.0",
+    "@nx/web": "^16.0.0",
+    "@nx/workspace": "^16.0.0",
```
Comment (supproduction) [inline package.json:82]: Can we use the same approach for all other new libs?

File: package.json [hunk 5]
```diff
@@ -66,73 +66,80 @@
...
+    "nx": "^16.0.0",
+    "nx-cloud": "16.4.0",
     "nx-stylelint": "^13.5.3",
     "postcss": "^8.4.16",
     "postcss-html": "^1.5.0",
     "postcss-syntax": "^0.36.2",
     "prettier": "2.7.1",
     "prettier-plugin-organize-imports": "^3.2.3",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
```
Comment (supproduction) [inline package.json:134]: Why do we need it?

File: package.json [hunk 6]
```diff
@@ -66,73 +66,80 @@
...
     "express": "^4.17.1",
     "husky": "^7.0.2",
     "jest-useragent-mock": "^0.1.1",
     "jsdom": "^20.0.0",
     "lerna": "^6.0.0",
     "mockdate": "^3.0.5",
     "npm-run-all": "^4.1.5",
-    "nx": "14.5.8",
+    "nx": "^16.0.0",
+    "nx-cloud": "16.4.0",
```
Comment (supproduction) [inline package.json]: Do we need nx-cloud?

File: package.json [hunk 7]
```diff
@@ -141,6 +148,7 @@
     "stylelint-no-unsupported-browser-features": "^5.0.3",
     "stylelint-order": "^5.0.0",
     "stylelint-use-logical-spec": "^3.2.2",
+    "tsconfig-paths-webpack-plugin": "^4.1.0",
```
Comment (supproduction) [inline package.json]: Why do we need it for webpack?

**Review by supproduction — CHANGES_REQUESTED** [review]

File: package.json (hunk 7 above)
Comment (tarashaliuk) [reply package.json]: because we use it Storybook. Will be removed when we will move to vite instead of webpack

File: package.json (hunk 6 above)
Comment (tarashaliuk) [reply package.json]: it is recommended as a caching tool that could speedup CI builds, but it shouldn't be in the scope of the current story. Will remove it

File: package.json (hunk 5 above)
Comment (tarashaliuk) [reply package.json:134]: Storybook 7 requires react

File: nx.json (hunk 3 above)
Comment (tarashaliuk) [reply nx.json]: It was added with nx-cloud. Will remove it

File: migrations.json
Comment (tarashaliuk) [reply migrations.json:1]: probably not. will remove it

File: libs/base/ui/overlays/modal/src/stories/interactive/close-modal.stories.ts (hunk 1 above)
Comment (tarashaliuk) [reply libs/base/ui/overlays/modal/src/stories/interactive/close-modal.stories.ts:31]: better to stay with await

**Pull request closed without merging**